includes/Stories_Renderer/Renderer.php

```diff
@@ -550,4 +591,81 @@ protected function get_content_overlay() {
 
 	}
 
+	/**
+	 * Renders the lightbox markup for non-amp pages.
+	 *
+	 * @return void
+	 */
+	protected function render_stories_with_lightbox_noamp() {
+		?>
+		<div class="web-stories-list__lightbox">
+			<amp-story-player width="3.6" height="6" layout="responsive">
+				<script type="application/json">
+				<?php
+				$data = [
+					'controls' => [
+						[
+							'name'     => 'close',
+							'position' => 'start',
+						],
+						[
+							'name' => 'skip-next',
+						],
+					],
+				];
+				echo wp_json_encode( $data );
+				?>
+				</script>
+				<?php echo wp_kses_post( $this->lightbox_html ); ?>
```

Comment:
Our own generated HTML should be safe to use without wp_kses, no?

(Aside: why first adding everything to `$this->lightbox_html` and then render that?)

Reply:
This was to save two loops while rendering the lightbox markup. Here we are collecting all the required markup ( all the `story links <a></a>` ) while iterating through stories. And render it when we need to add the lightbox markup without running the loop through all the stories again.